.github/workflows/test-cygwin.yml

```
@@ -76,17 +76,23 @@ jobs:
         with:
           dirs: 'C:\cygwin\bin;C:\cygwin\lib\lapack'
 
+      - name: Select Python version
+        run: |
+          ln -sf c:/cygwin/bin/python3.${{ matrix.python-minor-version }} c:/cygwin/bin/python3
+
+      - name: Get latest NumPy version
+        id: latest-numpy
+        shell: bash.exe -eo pipefail -o igncr "{0}"
+        run: |
+          python3 -m pip list --outdated | grep numpy | sed -r 's/ +/ /g' | cut -d ' ' -f 3 | sed 's/^/version=/' >> $GITHUB_OUTPUT
```

Perhaps use JSON output from `pip list`, so we're not dependant on the list output being stable? `jq` should already be installed for GitHub Actions.

```suggestion
          python3 -m pip list --outdated --format json | jq -r '.[] | select(.name == "numpy") | .version' >> $GITHUB_OUTPUT
```

---

It's not installed - https://github.com/radarhere/Pillow/actions/runs/5552959277/jobs/10140947590#step:7:8

What do you think of using Python like so?
`python3 -m pip list --outdated --format json | python3 -c 'import json, sys;packages = [package["latest_version"] for package in json.loads(sys.stdin.read()) if package["name"] == "numpy"];print("version="+(packages[0] if packages else ""))' >> $GITHUB_OUTPUT`

---

Oh, this simple test had `jq` preinstalled, but it's not doing all the other stuff `test-cygwin.yml` is:
https://github.com/hugovk/test/actions/runs/5552542172/jobs/10140017866#step:5:23

Anyway, the Python one is quite a mouthful, but probably fine too? I'll let you decide, either the Python or sed versions are fine :) Thanks for fixing this!

---

If you drop the `egor-tensin/cleanup-path` part on line 76, you should have `jq` available again.  That's there primarily to ensure the runner only finds Cygwin versions of things and so it says it can't find things if they aren't installed, instead of pulling in the Windows version and then giving an unrelated error.